src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.ConnectionManager.cs

```diff
@@ -114,6 +114,10 @@ private partial class ConnectionManager
                 var dataRpc = _remotableDataRpc.TryAddReference();
                 if (dataRpc == null)
                 {
+                    // TODO: If we used multiplex stream we wouldn't get to this state and we could always assume to have a connection
+                    // unless the service process stops working, in which case we should report an error and ask user to restart VS
+                    // (https://github.com/dotnet/roslyn/issues/40146)
```

Comment:
After doing more refactoring in the next PR I see it would be better to throw here instead of returning null. I'll follow up in the next PR and leave it as is here.